File: routeMap/app/src/test/java/ru/android/zheka/gmapexample1/home/TestHomeBindingModule.java
package ru.android.zheka.gmapexample1.home;

import android.content.Context;

import org.mockito.Mockito;

import dagger.Module;
import dagger.Provides;
import ru.android.zheka.coreUI.IActivity;
import ru.android.zheka.model.HomeModel;
import ru.android.zheka.vm.IPanelHomeVM;

@Module(includes = TestHomeModule.class)
public class TestHomeBindingModule {
    IPanelHomeVM mockVM = Mockito.mock (IPanelHomeVM.class);

    @Provides
    public IPanelHomeVM bindHomeVM(HomeModel model, IActivity view) {
        return mockVM;//new PanelHomeVM (Mockito.mock(IActivity.class), Mockito.mock(HomeModel.class));//mockVM;// Mockito.mock (IPanelHomeVM.class)
    }
    @Provides
    public HomeModel provideHomeModel(Context view) {
        return new HomeModel (view);
    }
}

File: routeMap/app/src/test/java/ru/android/zheka/gmapexample1/home/TestHomeBindingModule.kt
package ru.android.zheka.gmapexample1.home

import dagger.Module
import dagger.Provides
import org.mockito.Mockito
import ru.android.zheka.fragment.*
import ru.android.zheka.model.IHomeModel
import ru.android.zheka.vm.IPanelHomeVM
import ru.android.zheka.vm.PanelHomeVM

@Module(includes = [TestHomeModule::class])
class TestHomeBindingModule {
    var mockVM = Mockito.mock(IPanelHomeVM::class.java)

    @Provides
    fun bindHomeVM(model: IHomeModel?, view: IHome?, edit: IEdit, editTraces: IEditTraces, trace: ITrace, enterPoint: IEnterPoint)
            : IPanelHomeVM {
        return PanelHomeVM(view!!, model!!, edit, editTraces, trace, enterPoint)
    }

    @Provides
    fun provideEdit(): IEdit {
        return Edit()
    }

    @Provides
    fun provideEditTraces(): IEditTraces {
        return EditTraces()
    }

    @Provides
    fun provideTrace(): ITrace {
        return Trace()
    }

    @Provides
    fun provideEnterPoint(): IEnterPoint {
        return EnterPoint()
    }
}